src/lib/components/tabsv2/Tabsv2.component.js

```diff
@@ -46,27 +46,53 @@ function Tabs({
   const location = useLocation();
   const history = useHistory();
   const [selectedTabIndex, setSelectedTabIndex] = useState<?number>(null);
+  const queryURL = new URLSearchParams(location.search);
 
   const filteredTabsChildren = React.Children.toArray(children).filter(
     (child) => React.isValidElement(child) && child.type === Tab,
   );
 
+  const matchQuery = useCallback(
+    (query: Query): boolean => {
+      for (const key of Object.keys(query)) {
+        if (!(queryURL.has(key) && queryURL.get(key) === query[key]))
+          return false;
+      }
+      return true;
+    },
+    [queryURL],
+  );
+
+  const serialize = (query?: Query): string => {
```

Comment:
this is equivalent to `!query ? '' : '?'+ URLSearchParams.toString();`

Reply:
query[key] may need to be url encoded (you can try query params containing &test= as a value to test)